test/boost_optional.cpp

```diff
@@ -133,6 +133,32 @@ int main()
         BOOST_TEST(v2.get() == "abc");
     }
 
+    //optional.emplace(...)
+    {
+        struct NonMovable
+        {
+            NonMovable(NonMovable&&) = delete;
+            NonMovable& operator=(NonMovable&&) = delete;
+
+            NonMovable() = default;
+            NonMovable(std::string str) : str{std::move(str)}
+            {}
+
+            std::string str;
+        };
+
+        coruja::optional<NonMovable> v;
```

Comment:
I don't understand why we need a `value_type` that isn't movable to perform a basic test of `emplace`. Isn't there enough to use `coruja::optional<std::string>` instead of `coruja::optional<NonMovable>`?

Reply:
I chose to test `emplace` with a non-movable `value_type` because this is a common motivation for the use of `emplace` (it's what motivated me create this patch in the first place). So I believe it would be sensible to make this the default test case for `emplace`.